### PR TITLE
Move rustc-guide submodule to rustc-dev-guide

### PR DESCRIPTION
**Pull request by spastorino**
r? @pietroalbini

### CODE REVIEW

**Comment by pietroalbini** [comment]
While the submodule part of this PR is good, extra care should be taken on changing toolstate. There are two requirements on my end for such change:

* It shouldn't require opening PRs in the toolstate repository to manually change the data: the toolstate system should be able to detect the new tool doesn't exist yet and add it.
* It can't break PR builds of branches created before this is merged: this means the `mingw-check` builder should pass with the old validation code and the new `latest.json`.

I'm not *deeply* familiar with toolstate (cc @kennytm @Mark-Simulacrum), and I need to think a bit more what the best solution for this is. I'll post other comments when I think of something.

**Comment by spastorino** [comment]
We also have this around https://github.com/rust-lang-nursery/rust-toolstate/pull/16. I know that you are aware @pietroalbini, just sharing for the rest :).

**Comment by mark-i-m** [comment]
> It shouldn't require opening PRs in the toolstate repository to manually change the data: the toolstate system should be able to detect the new tool doesn't exist yet and add it.

IIRC, this was not the case when we first added rustc-dev-guide toolstate

**Comment by mark-i-m** [comment]
I think we can fix this now, correct?

**Comment by mark-i-m** [comment]
ping @rust-lang/infra Would be great to fix this soon...

**Comment by kennytm** [comment]
Blocked by proper fix in rust-lang-nursery/rust-toolstate#16.

File: src/tools/publish_toolstate.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
         'ryankurte', 'thejpster', 'therealprof',
     },
     'edition-guide': {'ehuss', 'Centril', 'steveklabnik'},
-    'rustc-guide': {'mark-i-m', 'spastorino', 'amanjeev', 'JohnTitor'},
+    'rustc-dev-guide': {'mark-i-m', 'spastorino', 'amanjeev', 'JohnTitor'},
```
Comment (kennytm) [inline src/tools/publish_toolstate.py]: either keep the `rustc-guide` maintainers or change line 206 to use `MAINTAINERS.get(tool, ())`.

Comment (spastorino) [reply src/tools/publish_toolstate.py]: Done!

**Comment by crlf0710** [comment]
@rustbot modify labels to -S-blocked +S-waiting-on-author

Now that rust-lang-nursery/rust-toolstate#16 is merged, this is unblocked. Maybe do a rebase to see whether the CI becomes green? @spastorino

**Comment by Xanewok** [comment]
Unfortunately this seems to have broken the entire tree - everything eagerly fails on mingw-check job.

Applying this patch (#70149) still yields:
```
 Cloning into 'rust-toolstate'...
  File "../../src/tools/publish_toolstate.py", line 208
    if new > old:
                ^
SyntaxError: invalid syntax
```

File: src/tools/publish_toolstate.py [hunk 2]
```diff
@@ -204,7 +204,7 @@ def update_latest(
                 old = status[os]
                 new = s.get(tool, old)
                 status[os] = new
-                maintainers = ' '.join('@'+name for name in MAINTAINERS[tool])
+                maintainers = ' '.join('@'+name for name in MAINTAINERS.get(tool, ())
```
Comment (mati865) [inline src/tools/publish_toolstate.py:207]:
```suggestion
                maintainers = ' '.join('@'+name for name in MAINTAINERS.get(tool, ()))
```
Missing `)`

**Comment by Xanewok** [comment]
@mati865 nice catch! Let's land this separately.

@bors r+ p=100 (please adjust accordingly if that's not okay, just want to unblock the tree)

**Comment by mati865** [comment]
`mingw-check` is green now.

**Comment by mark-i-m** [comment]
Thanks @mati865 and @Xanewok !

**Comment by Xanewok** [comment]
@Dylan-DPC I guess my power level is too low 😢

**Comment by mati865** [comment]
@Xanewok bors doesn't work with closed PRs. Next time you want to open the tree do it in some open PR 😉

**Comment by pnkfelix** [comment]
removing beta-nominated tag, as this was merged into beta channel as part of PR #70848

**Pull request merged**